array_test.go

```diff
@@ -0,0 +1,69 @@
+package protobuf
+
+import (
+	"fmt"
+	"testing"
+	"github.com/stretchr/testify/assert"
+	"reflect"
+	"encoding/hex"
+)
+
+type ArrayTest0 struct {
+	A []int
+}
+
+type ArrayTest1 struct {
+	A []int64
+}
+
+type ArrayTest2 struct {
+	A []int32
+}
+
+type ArrayTest3 struct{
+	A int
+}
+
+
+func TestArray(t *testing.T){
+
+	fmt.Println("TestArray:")
+
+	// largest int32 is 2147483647
+	var large int = 3147483647
+
+	a0 := ArrayTest0{[]int{1, 1, large}}
+	a1 := ArrayTest1{[]int64{1, 1, 1}}
+	a2 := ArrayTest2{[]int32{1, 1, 1}}
+	a3 := ArrayTest3{1}
+
+	buf0, _ := Encode(&a0)
+	buf1, _ := Encode(&a1)
+	buf2, _ := Encode(&a2)
+	buf3, _ := Encode(&a3)
+
+	t.Log(hex.Dump(buf0))
+	t.Log(hex.Dump(buf1))
+	t.Log(hex.Dump(buf2))
+	t.Log(hex.Dump(buf3))
+
+
+	b0 := ArrayTest0{}
+	b1 := ArrayTest1{}
+	b2 := ArrayTest2{}
+	b3 := ArrayTest3{}
+
+	Decode(buf0, &b0)
+	t.Log(b0, reflect.TypeOf(b0))
+
+	Decode(buf1, &b1)
+	t.Log(b1, reflect.TypeOf(b1))
+
+	Decode(buf2, &b2)
+	t.Log(b2, reflect.TypeOf(b2))
+
+	Decode(buf3, &b3)
+	t.Log(b3, reflect.TypeOf(b3))
+
+	assert.True(t, true, nil)
```

Comment:
You should actually test that the returned arrays are the same ;) Something like:
```
require.Equal(t, a0, b0)
require.Equal(t, a1, b1)
require.Equal(t, a2, b2)
require.Equal(t, a3, b3)
```
Even though I'm never sure if `require.Equal` works with arrays...

Reply:
Thanks,
apparently https://github.com/stretchr/testify/issues/275 says that assert.Equal (or require) should work fine with arrays if they are sorted similarly, so that should be fine.